hail/python/hailtop/aiotools/fs/fs.py

```diff
@@ -46,9 +46,11 @@ def name(self) -> str:
     async def url(self) -> str:
         pass
 
-    @abc.abstractmethod
-    def url_maybe_trailing_slash(self) -> str:
-        pass
+    async def url_maybe_trailing_slash(self) -> str:
+        return await self.url()
+
+    async def url_with_query(self) -> str:
+        return await self.url()
```

Comment:
Huh, I feel like all three of these methods should not be `async`. Instead of making these async can you make `url` sync?